sycl/source/device_selector.cpp

```diff
@@ -97,23 +111,10 @@ device device_selector::select_device() const {
 /// 1. GPU
 /// 2. CPU
 /// 3. Host
+/// 4. Accelerator
 int default_selector::operator()(const device &dev) const {
-
   int Score = REJECT_DEVICE_SCORE;
 
-  // Give preference to device of SYCL BE.
-  if (isDeviceOfPreferredSyclBe(dev))
-    Score = 50;
-
-  // If SYCL_DEVICE_FILTER is set, filter device gets a high point.
-  // All unmatched devices should never be selected.
-  detail::device_filter_list *FilterList =
-      detail::SYCLConfig<detail::SYCL_DEVICE_FILTER>::get();
-  // device::get_devices returns filtered list of devices.
-  // Keep 1000 for default score when filters were applied.
-  if (FilterList)
-    Score = 1000;
```

Comment:
Why is this removed? I am not completely sure, but it seems like this tries to ensure that no device is rejected if the user applied a filter. Maybe we could simply change it to a `+= 1` to ensure this?